### ISSUE TITLE
Replace `development` flag with runtime compile options

### ISSUE THREAD

**Issue by srid** [Plutonomicon/plutarch-plutus#359]
We want to evaluate Plutarch programs with tracing enabled on _specific_ tests (for debugging purposes) without enabling it globally elsewhere (doing so would adversely affect the golden tests).

The CPP approach wouldn't work because it enables it globally.

Therefore, to solve this, I propose that `compile` take *runtime* options, deciding to enable tracing or not (these options can also eventually represent #260)

```haskell
compile' :: CompileOpts -> ClosedTerm a -> Script
compile' = ...

data CompileOpts = CompileOpts 
  { enableTracing :: Bool 
  }
```

/cc @V0ldek 

---

How do we implement it? It looks like we will end up needing #260 here.

**Comment by Saizan** [comment]
I do not think we need #260 here. It would be a matter of `Term` taking an extra `CompileOpts` argument under the hood, so that `ptrace` can case on it.

**Comment by L-as** [comment]
Will do this ASAP along with the soundness fixes once I deal with #443
